src/types/token.type.ts

```diff
@@ -1,11 +1,17 @@
-export type Token = {
+import { ERC20Instance } from '@rsksmart/rif-relay-contracts/types/truffle-contracts';
+
+type TokenAttributes = {
     name: string;
-    decimals?: number;
-    symbol?: string;
-    contractAddress: string;
+    decimals: number;
+    symbol: string;
+};
+
+export type ERC20Token = Partial<TokenAttributes> & {
+    instance: ERC20Instance;
 };
 
-export type TokenOptions = {
-    decimals?: boolean;
-    symbol?: boolean;
+type OptionsFlags<Type> = {
+    [Property in keyof Type]: boolean;
 };
+
+export type ERC20Options = Partial<OptionsFlags<TokenAttributes>>;
```

Comment:
```suggestion
type TokenAttributePropertyName = keyof TokenAttributes;

export type ERC20Options = Partial<Record<TokenAttributePropertyName, boolean>>;
```

Reply:
updated

Reply:
Not sure about the changes you applied regarding this.

Reply:
you were right, the push did not went thought.